ophyd/ophydobj.py

```diff
@@ -34,22 +35,45 @@ class StatusBase:
     """
     This is a base class that provides a single-slot
     call back for finished.
+
+    Parameters
+    ----------
+    timeout : float, optional
+        The default timeout to use for a blocking wait, and the amount of time
+        to wait to mark the operation as failed
     """
-    def __init__(self):
+    def __init__(self, *, timeout=None):
         super().__init__()
         self._lock = RLock()
         self._cb = None
         self.done = False
         self.success = False
+        self.timeout = None
+
+        if timeout is not None:
+            self.timeout = float(timeout)
+
+        if self.timeout is not None:
```

Comment:
Then suggest something better. Or perhaps wait (as we discussed in the chat) to talk about the overall implications together rather than the little things that bug you.